sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/E2E.Tests/AzureMonitorTraceExporterLiveTests.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Monitor.OpenTelemetry.Exporter.E2E.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using System.Threading.Tasks;
+
+    using Azure.Core.TestFramework;
+    using Azure.Monitor.OpenTelemetry.Exporter;
+    using Azure.Monitor.OpenTelemetry.Exporter.E2E.Tests.TestFramework;
+    using Azure.Monitor.Query;
+    using Azure.Monitor.Query.Models;
+    using global::OpenTelemetry;
+    using global::OpenTelemetry.Resources;
+    using global::OpenTelemetry.Trace;
+
+    using NUnit.Framework;
+
+    /// <summary>
+    /// Collection of tests to evaluate the <see cref="AzureMonitorLogExporter"/>.
+    /// </summary>
+    public class AzureMonitorTraceExporterLiveTests : RecordedTestBase<MonitorExporterTestEnvironment>
+    {
+        public AzureMonitorTraceExporterLiveTests(bool isAsync) : base(isAsync) { }
+
+        [RecordedTest]
+        [PlaybackOnly("Add field level validations")]
+        public async Task VerifyTraceExporter()
+        {
+            // SET UP
+            ActivitySource src = new ActivitySource("TestActivitySource");
+            string roleName = nameof(VerifyTraceExporter);
+
+            var resourceAttributes = new Dictionary<string, object>
+            {
+                { "service.name", roleName },
+            };
+
+            var resourceBuilder = ResourceBuilder.CreateDefault().AddAttributes(resourceAttributes);
+
+            using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+                .SetResourceBuilder(resourceBuilder)
+                .AddSource("TestActivitySource")
+                .SetSampler(new AlwaysOnSampler())
+                .AddAzureMonitorTraceExporter(o => o.ConnectionString = TestEnvironment.ConnectionString)
+                .Build();
+
+            using (var activity = src.StartActivity("Test", ActivityKind.Internal))
+            {
+                activity?.SetTag("foo", "bar");
+            }
+
+            // Export
+            tracerProvider.ForceFlush();
+
+            // Query
+            var client = CreateClient();
+
+            string query = $"AppDependencies | where AppRoleName == '{roleName}'";
+
+            LogsTable table = await CheckForRecord(client, query);
+
+            // Assert
+            Assert.True(table.Rows.Count() > 0);
+        }
+
+        private async Task<LogsTable> CheckForRecord(LogsQueryClient client, string query)
```

Comment:
Consider including a Guid in the telemetry item. Maybe `activity.SetTag` ?
The point is that if you run this test multiple times, the verify would fail.

Reply:
I am keeping the roleName unique. I know that can fail too if the tests is run in live mode multiple times as there will be more than one record matching the roleName. Right now, the assumption is this will be run only once (as per my understanding live tests are run only during nightly build)

I was trying to use guid but then it needs special handling when the test is run in playback mode as the query will fail (we will have a new guid every time the test runs) and there will be no recorded data that will match this newly created guid. I also tried doing special handling in case of playback versus recorded mode (e.g. using `like` versus `==` in `where` clause) it did not work as expected. This is one of the things I need to look in to :)

Reply:
I have used roleName for uniqueness. To resolve case of duplicates getting top(1) by timegenerated.